include/cocaine/api/connect.hpp

```diff
@@ -72,8 +66,11 @@ template<class Tag>
 class client:
     public details::basic_client_t
 {
+    template<class Event, bool = std::is_same<typename Event::tag, Tag>::value>
+    struct traits;
+
     template<class Event>
-    struct traits {
+    struct traits<Event, true> {
```

Comment:
What's the benefit of weird compilation error on static assert with pretty looking message?